### ISSUE TITLE
`Clang` can be constructed without `Clang::new()`

### ISSUE THREAD

**Issue by io12** [KyleMayes/clang-rs#35]
It's possible to construct `Clang` with

```rust
let cl = clang::Clang;
```

instead of

```rust
let cl = clang::Clang::new().unwrap();
```

This violates the requirement that only one `Clang` object exists per thread. The way to fix it is to give the struct a private inner type.

```rust
struct ClangInner; // Private, so can't construct `Clang` without `Clang::new()`

pub struct Clang(ClangInner);
```

**Comment by KyleMayes** [comment]
I just released `v1.0.0` and already need to make a breaking change for this and #26...

I guess I needed to mark the commonly updated enums as non-exhaustive so every new Clang version doesn't result in a breaking change anyway, so hopefully a `v2.0.0` will come out with this fix soon.

**Comment by io12** [comment]
I'd argue that it isn't *really* a breaking change, since constructing `Clang` without `new()` is using the crate incorrectly. It restricts what you can do with the crate, but it doesn't restrict the correct use of the crate.
